pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -328,13 +329,17 @@ func (m *manager) updateContainerCPUSet(p *v1.Pod, containerID string, cpus cpus
 			})
 	}
 
-	runtimeService, err := m.runtimeManager.GetRuntimeServiceByPod(p)
+	podWorkloadType := runtimeregistry.ContainerWorkloadType
+	if p.Spec.VirtualMachine != nil {
+		podWorkloadType = runtimeregistry.VmworkloadType
+	}
```

Comment:
This piece of logic should be very common.  Consider having a wrapper method. But later on if we support mixed pod (one pod contains both VMs and containers) that wrapper will not work :)